model/src/contract.rs

```diff
@@ -286,13 +286,43 @@ where
     }
 }
 
+/// A return status code for `deploy` and `dispatch` calls back to the SRML contracts module.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct RetCode(u8);
+
+impl RetCode {
+    /// Indicates a successful execution.
+    pub fn success() -> Self {
+        Self(0)
+    }
+
+    /// Indicates a failure upon execution.
+    pub fn failure() -> Self {
+        Self(255)
+    }
+
+    /// Returns the internal `u32` value.
+    pub fn to_u32(self) -> u32 {
+        self.0 as u32
+    }
+}
+
 /// A simple interface to work with contracts.
 pub trait Contract {
     /// Deploys the contract state.
     ///
     /// Should be performed exactly once during contract lifetime.
     /// Consumes the contract since nothing should be done afterwards.
-    fn deploy(self);
+    ///
+    /// # Note
+    ///
+    /// Only the least-significant 8 bits of the `u32` return value may
+    /// be used to indicate success or an error state.
+    /// Zero (`0`) represents a successful execution, (`255`) means invalid
+    /// execution (e.g. trap) and any value in between represents a non-
+    /// specified invalid execution.
+    /// All other bits in the `u32` must be 0.
```

Comment:
This comment is redundant now that `RetCode` enforces it. Perhaps could move it (and the duplicate on deploy) to the `RetCode` type?